### ISSUE TITLE
@typescript-eslint/no-shadow: false positive in markdown code fences

### ISSUE THREAD

**Issue by privatenumber** [privatenumber/lintroll#100]
## Error
```
/Users/osame/Developer/github/privatenumber/.claude/rules/testing.md
  35:36  error    'test' is already declared in the upper scope on line 1 column 20       @typescript-eslint/no-shadow
```

## Code
The error is triggered by this TypeScript code fence in a markdown file:

~~~ts
import { describe, test, expect } from 'manten'
import { createFixture } from 'fs-fixture'

await describe('pkgroll build', ({ test }) => {
    await test('generates ESM output', async ({ onTestFail }) => {
        // ...
    })
})
~~~

## Problem
This is a **false positive** for two reasons:

1. **Markdown code fences are documentation examples**, not real code being executed. They should have relaxed linting rules.

2. **The `no-shadow` allow list is conditional on manten being installed** (`isInstalled('manten')` in `eslint.ts`). Since this is a documentation project without manten as a dependency, the `test` variable isn't in the allow list - even though the code is demonstrating manten usage.

The `test` parameter in manten's `describe` callback is **idiomatic usage** - it's supposed to shadow the import to provide a scoped `test` function.

## Current Behavior
`@typescript-eslint/no-shadow` errors on code fences demonstrating testing patterns, even though:
- The code is documentation, not real code
- The pattern being demonstrated is correct and idiomatic

## Suggested Fix
Add `@typescript-eslint/no-shadow: 'off'` to the markdown TypeScript config in `src/configs/markdown.ts`:

```ts
defineConfig({
    files: ['**/*.md/*.{ts,tsx}'],
    rules: {
        // ... existing rules ...
        '@typescript-eslint/no-shadow': 'off',
    },
}),
```

This follows the pattern already established for other strict rules that are relaxed for markdown code fences (e.g., `@typescript-eslint/no-unused-vars: 'warn'`).
